File: apps/mybookkeeper/frontend/src/shared/types/welcome-manual/welcome-manual-summary.ts
/**
 * Mirrors backend `WelcomeManualSummary` Pydantic schema (GET /welcome-manuals
 * list items). ``section_count`` is computed server-side, not a column.
 */
export interface WelcomeManualSummary {
  id: string;
  title: string;
  property_id: string | null;
  section_count: number;
  created_at: string;
  updated_at: string;
}

File: apps/mybookkeeper/backend/app/schemas/welcome_manuals/welcome_manual_summary.py
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class WelcomeManualSummary(BaseModel):
    """Minimal manual payload for the list view.

    ``section_count`` is not a column — the service populates it from a single
    grouped count query, so this model is constructed explicitly rather than
    via ``model_validate``.
    """

    id: uuid.UUID
    title: str
    property_id: uuid.UUID | None = None
    section_count: int = 0
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)
